vectordraw/vectordraw.py

```diff
@@ -0,0 +1,544 @@
+            assert field.scope in (Scope.content, Scope.settings), (
+                "Only Scope.content or Scope.settings fields can be used with "
+                "StudioEditableXBlockMixin. Other scopes are for user-specific data and are "
+                "not generally created/configured by content authors in Studio."
+            )
+            field_info = self._make_field_info(field_name, field)
+            if field_info is not None:
+                context["fields"].append(field_info)
+        fragment.add_content(loader.render_template("templates/html/vectordraw_edit.html", context))
+        # Add resources to studio_view fragment
+        fragment.add_css_url(
+            self.runtime.local_resource_url(self, 'public/css/vectordraw.css')
+        )
+        fragment.add_css_url(
+            self.runtime.local_resource_url(self, 'public/css/vectordraw_edit.css')
+        )
+        fragment.add_javascript_url(
+            "//cdnjs.cloudflare.com/ajax/libs/jsxgraph/0.98/jsxgraphcore.js"
+        )
+        fragment.add_javascript_url(
+            self.runtime.local_resource_url(self, 'public/js/studio_edit.js')
+        )
+        fragment.add_javascript_url(
+            self.runtime.local_resource_url(self, 'public/js/vectordraw_edit.js')
+        )
+        fragment.initialize_js(
+            'VectorDrawXBlockEdit', {"settings": self.settings}
+        )
+        return fragment
+
+    def validate_field_data(self, validation, data):
+        """
+        Validate this block's field data.
+        """
+        super(VectorDrawXBlock, self).validate_field_data(validation, data)
+
+        def add_error(msg):
+            """ Helper function for adding validation messages. """
+            validation.add(ValidationMessage(ValidationMessage.ERROR, msg))
+
+        if data.background_url.strip():
+            if data.background_width == 0 and data.background_height == 0:
+                add_error(
+                    u"You specified a background image but no width or height. "
+                    "For the image to display, you need to specify a non-zero value "
+                    "for at least one of them."
+                )
+            if not data.background_description.strip():
+                add_error(
+                    u"No background description set. "
+                    "This means that it will be more difficult for non-visual users "
+                    "to solve the problem. "
+                    "Please provide a description that contains sufficient information "
+                    "that would allow anyone to solve the problem if the image did not load."
+                )
+
+    def _validate_check_answer_data(self, data):  # pylint: disable=no-self-use
+        """
+        Validate answer data submitted by user.
+        """
+        # Check vectors
+        vectors = data.get('vectors')
+        if not isinstance(vectors, dict):
+            return False
```

Comment:
Might be better to have this function raise a `ValueError` when it finds an issue, rather than returning False.